core/frontend/src/BasicUnitsProvider.ts

```diff
@@ -17,19 +17,28 @@ import {
  */
 export class BasicUnitsProvider implements UnitsProvider {
   /** Find a unit given the unitLabel. */
-  public async findUnit(unitLabel: string, unitFamily?: string): Promise<UnitProps> {
+  public async findUnit(unitLabel: string, unitFamily?: string, unitSystem?: string): Promise<UnitProps> {
+    const labelToFind = unitLabel.toLowerCase();
+    const unitFamilyToFind = unitFamily ? unitFamily.toLowerCase():undefined;
+    const unitSystemToFind = unitSystem ? unitSystem.toLowerCase():undefined;
+
     for (const entry of UNIT_DATA) {
       if (unitFamily) {
-        if (entry.unitFamily !== unitFamily)
+        if (entry.unitFamily.toLowerCase() !== unitFamilyToFind)
```

Comment:
nit: Can you put these two if statements on the same line instead of breaking it up?

Reply:
fixed.